src/solvers/td3.ml

```diff
@@ -122,6 +122,16 @@ module WP =
       let rho_write = data.rho_write in
       let dep = data.dep in
 
+      let skip_unchanged_rhs =
+        let enabled = GobConfig.get_bool "solvers.td3.skip-unchanged-rhs" in
+        if enabled && (restart_sided || restart_only_globals || restart_only_access || restart_wpoint || restart_once) then
+          (M.warn "restarting active, disabling solvers.td3.skip-unchanged-rhs"; false)
+        else
+          enabled
+      in
+      (* TODO: Do we want to marshal dep_vals? *)
+      let dep_vals = HM.create 10 in
```

Comment:
I think the issue is that destabilization no longer re-triggers all side-effects which we rely on when restarting e.g. globals.

Reply:
If something is restarted, then the value in `rho` is changed, so the checking should detect that and inevitably re-evaluate the right-hand side to retrigger all the side effects. I'm just surprised because in #364 this wasn't a problem, as far as I remember.

Or at worst, if it somehow is, then maybe just disable this for the duration of restarting/postsolving. Because with restarting enabled, normal solving should still be able to benefit from this.

Reply:
>  so the checking should detect that and inevitably re-evaluate the right-hand side to retrigger all the side effect

It might be true in all our examples, but consider a global that is re-started and an unknown that has a side-effect TO it. Thus unknown is de-stabilized, but then we find that none of its dependencies has changed and the RHS is not re-evaluated, right? So we'd be missing a side-effect?!